cmd/util/cmd/execution-state-extract/execution_state_extract_test.go

```diff
@@ -207,6 +213,312 @@ func TestExtractExecutionState(t *testing.T) {
 	})
 }
 
+// TestExtractPayloadsFromExecutionState tests state extraction with checkpoint as input and payload as output.
+func TestExtractPayloadsFromExecutionState(t *testing.T) {
+
```

Comment:
```suggestion
	t.Parallel()
```

Reply:
Adding `t.Parallel()` here fails test because output dictionary content is not as expected during tests.

Reply:
Ah, OK. Guess that stateful-ness needs to be fixed in a separate PR